public/app/features/dashboard-scene/scene/LibraryPanelBehavior.tsx

```diff
@@ -123,6 +123,8 @@ export class LibraryPanelBehavior extends SceneObjectBase<LibraryPanelBehaviorSt
 
     try {
       const libPanel = await getLibraryPanel(this.state.uid, true);
+      // Keep title from dashboard JSON's panel model
+      libPanel.model.title = this.state.title ?? libPanel.model.title;
```

Comment:
is there corresponding logic in the old arch? how is this title set?

Reply:
As you mentioned, here it is: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/PanelModel.ts#L690

The first taken title is from the model, so if the library panel doesn't have the property, the title from the model is used.